packages/server/test/unit/config_spec.js

```diff
@@ -26,40 +19,6 @@ describe('lib/config', () => {
     process.env = this.env
   })
 
-  context('environment name check', () => {
-    it('throws an error for unknown CYPRESS_INTERNAL_ENV', async () => {
-      sinon.stub(errors, 'throwErr').withArgs('INVALID_CYPRESS_INTERNAL_ENV', 'foo-bar')
-      process.env.CYPRESS_INTERNAL_ENV = 'foo-bar'
-      const cfg = {
-        projectRoot: '/foo/bar/',
-        supportFile: false,
-      }
-      const options = {}
-
-      try {
-        await config.mergeDefaults(cfg, options)
-      } catch {
-        //
-      }
-
-      expect(errors.throwErr).have.been.calledOnce
-    })
-
-    it('allows production CYPRESS_INTERNAL_ENV', async () => {
-      sinon.stub(errors, 'throwErr')
-      process.env.CYPRESS_INTERNAL_ENV = 'production'
-      const cfg = {
-        projectRoot: '/foo/bar/',
-        supportFile: false,
-      }
-      const options = {}
-
-      await config.mergeDefaults(cfg, options)
-
-      expect(errors.throwErr).not.to.be.called
-    })
-  })
-
   context('.get', () => {
```

Comment:
The tests under `.get` more suit `pkg/data-context/test/unit/data/ProjectLifecycleManager`. 

If you agree, I'll move them in the next PR.